### PR TITLE
Add pre-existing cprover utils to the new repository. 

### PR DESCRIPTION
**Pull request by thk123**
Add CProver from models-library:
  model/src/main/java/org/cprover/CProver.java

Add CProverString from models-library:
  model/src/main/java/org/cprover/CProverString.java

Add IgnoredMethodImplementation from models-library:
  model/src/main/java/org/cprover/IgnoredMethodImplementation.java

Added MustNotThrow from models-library:
  model/src/main/java/org/cprover/MustNotThrow.java

Added OverlayClassImplementation from models-library:
  model/src/main/java/org/cprover/OverlayClassImplementation.java

Added OverlayMethodImplementation from models-library:
  model/src/main/java/org/cprover/OverlayMethodImplementation.java

Added PreferredConcreteImplementation from models-library:
  model/src/main/java/comd/diffblue/annotation/PreferredConcreteImplementation.java

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <groupId>org.cprover</groupId>
+    <artifactId>util</artifactId>
+    <version>1.0-SNAPSHOT</version>
+
+
```
Comment (peterschrammel) [inline pom.xml]: redundant blank line

File: src/main/java/org/cprover/util/CProverString.java [hunk 2]
```diff
@@ -0,0 +1,422 @@
...
+        CProver.assume(value != null);
+        CProver.assume(value.length - count >= offset
+                && offset >= 0 && count >= 0);
+        StringBuilder builder = new StringBuilder();
+        for(int i = 0; i < count; i++) {
+            builder.append(value[offset + i]);
+        }
+        return builder.toString();
+    }
+}
```
Comment (peterschrammel) [inline src/main/java/org/cprover/util/CProverString.java]: missing new line

File: src/main/java/org/cprover/util/CProver.java [hunk 3]
```diff
@@ -0,0 +1,309 @@
+package org.cprover.util;
```
Comment (peterschrammel) [inline src/main/java/org/cprover/util/CProver.java]: License header missing:

```
// Copyright 2018 Diffblue Ltd. All rights reserved.
// Use of this source code is governed by a 3-clause BSD license that can be
// found in the LICENSE file.
```

Comment (thk123) [reply src/main/java/org/cprover/util/CProver.java]: Does this want to go into every Java file?

**Comment by thk123** [comment]
@peterschrammel comments [addressed](https://github.com/diffblue/cprover-java-util/compare/f34db28db8569f8a4bb97151c4fe7117557d02af..40a4c01d1fc5241cdee7becdf1f6d8a7d24173a6)

File: src/main/java/org/cprover/util/CProver.java (hunk 3 above)
Comment (peterschrammel) [reply src/main/java/org/cprover/util/CProver.java]: Yes

Comment (thk123) [reply src/main/java/org/cprover/util/CProver.java]: I assumed as much and did exactly that

**Pull request merged**